### PR TITLE
add manual meshing to preprocessing

### PR DESCRIPTION
**Pull request by cjdoolittle**

### CODE REVIEW

File: docs/source/preprocessing/manualMeshing.rst [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    -   The CGNS tree structure should be of the form base > block > boundary
+
+        -   Multiple blocks (domains) should be at the same level, 2\ :sup:`nd`\
+        -   Multiple boundaries (no-slip walls) should be at the same level, 3\ :sup:`rd`\ , within their respective blocks
+        -   Block interfaces should be split so that one interface is contained within each adjacent block
+
+    -   A Flow360Mesh.json file is preferable to define no-slip boundaries
+    -   Boundary names will be strings for CGNS meshes
+
+        -   If multi-block mesh, format will be <block-name>/<boundary-name>
```
Comment (maciej-flexcompute) [inline docs/source/preprocessing/manualMeshing.rst:174]: I think for a single block CGNS mesh is the same <block-name>/<boundary-name>

Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:179]: corrected

File: docs/source/preprocessing/manualMeshing.rst [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+The purpose of any mesh is to provide a stencil for calculation of the discretized governing equations of the flowfield. Stencil spacings, often referred to as mesh refinement, and structure, often characterized by element types, are important to CFD because they influence the numerical solution. That is, the refinement and type of mesh used dictates what flow physics can be solved and how accurate those flow physics are. Keep in mind that the flow solution is discretized on the mesh, not the underlying geometry.
+
+To that end, when considering a CFD analysis and building mesh(es), the user should carefully consider what flow physics are important to the final analysis objectives. Meshes should be purpose built to adequately capture flow features of interest and those impactful to critical analysis results.
+
+Questions to ask oneself:
+
+-   Where are large gradients (boundary layers, tip vortices, shocks, transition) expected in the flowfield?
+-   What regions of the model (wing, control surface, wake) are most important to the desired results?
+-   What are the operating conditions (Mach, altitude, attitude, rotation rates) and are they expected to vary?
```
Comment (jbungenerflex) [inline docs/source/preprocessing/manualMeshing.rst:27]: replace "attitude" with "alpha, beta"?

File: docs/source/preprocessing/manualMeshing.rst [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+        -   See flow360client.noSlipWallsFromMapbc() in :ref:`Python API Reference<api>`
+
+    -   Alternatively, a Flow360Mesh.json file can define no-slip boundaries
+    -   Boundary names will be integers for UGRID meshes
+    -   UGRID meshes are not appropriate for scenarios with multi-block motion
+
+-   CGNS considerations:
+
+    -   CGNS mesh should be export as an HDF5 file type
```
Comment (jbungenerflex) [inline docs/source/preprocessing/manualMeshing.rst:168]: TYPO: exportED

File: docs/source/preprocessing/preprocessing.rst [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+Preprocessing
+=============
+
+The following section focuses on preparing a mesh for Flow360 either manually, using meshing software, or automatically, using automated meshing functionality. Flow360 solver supports the following mesh formats:
+
+- AFLR3 (.ugrid).
+- CGNS (single and multi-block).
+
+for details of mesh formats see :ref:`here<manualMeshingFlow360Section>`.
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/preprocessing.rst:11]: allowable types are described in Flow360 section, not Volume section

File: docs/source/preprocessing/meshGuidelines.csv [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿Geometry,,Reference,Criteria,Source
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/meshGuidelines.csv:1]: formatting is not as intended (CSV restrictions). attempted to reformat.

File: docs/source/preprocessing/preprocessing.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (jbungenerflex) [reply docs/source/preprocessing/preprocessing.rst:11]: I am not sure I understand, the piece discussing allowed cell types: 
"Tetrahedral (tet), pyramid (pyr), prism (pri), and hexahedral (hex) elements are allowable

Elements must be first-order"

is under "3.1.3.2 Volume"
I guess confusion comes from the difference between cell types vs mesh types?

File: docs/source/preprocessing/meshGuidelines.csv [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+component,feature,,,intention
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/meshGuidelines.csv:2]: see above

File: docs/source/preprocessing/manualMeshing.rst [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+        -   Reference the max surface element length
+        -   Apply :math:`maxSurfaceElementLength \cdot \sqrt{3}` sizing for refinement regions near surfaces
+        -   Larger refinement regions can be extended further downstream, while still enclosing smaller regions, with incrementally larger sizing applied
+
+.. _manualMeshingFlow360Section:
+
+Flow360
+-------
+
+-   Boundary conditions (BCs) should be specified as the mesh is generated
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:139]: done

File: docs/source/preprocessing/manualMeshing.rst (hunk 3 above)
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:168]: fixed

File: docs/source/preprocessing/manualMeshing.rst [hunk 8]
```diff
@@ -0,0 +1,279 @@
...
+===============
+
+The following are recommended best-practices for generating suitable mesh for CFD. Recommendations are intentionally general to allow for users to develop meshes with their own tools and processes.
+
+See the following section, :ref:`Targets<Targets>`, for element sizing and quality metrics.
+
+Surface
+-------
+
+-   Triangular (tri) and quadrilateral (quad) elements are allowable, all tri is preferable
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:40]: removed. "preferable" in that the user does not need to manage all of the tri_<bc> & quad_<bc> conventions with JSON files. ANSA exports of CGNS can circumvent this, but I have not seen Pointwise be able to combine element types.

[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:40]: also, from my experience, all tri tends to help focus evaluation of mesh quality metrics and troubleshooting problematic mesh regions.

File: docs/source/preprocessing/manualMeshing.rst [hunk 9]
```diff
@@ -0,0 +1,279 @@
...
+    -   TE (blunt): multiple elements should be present between parallel edges
+
+-   Farfield:
+
+    -   Encloses entire fluid domain where flow solution occurs
+    -   Size to roughly :math:`\text{100x}` largest dimension of no-slip surfaces, centered around geometry
+    -   Edges should have roughly :math:`20` nodes in all directions, target isotropic elements
+    -   Depending on the intended boundary condition(s), a sphere (farfield) or box (inlet-outlet) are typically preferred
+
+-   Rotating interfaces:
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:69]: can update when tutorials completed and implemented in docs

File: docs/source/preprocessing/manualMeshing.rst [hunk 10]
```diff
@@ -0,0 +1,279 @@
...
+
+    -   On flat, circular surfaces nodes **must** be positioned on concentric rings
+    -   On cylindrical surfaces extruding the outer circular edge/nodes with consistent spacing is preferable, target element size for isotropic elements
+    -   Should be roughly :math:`\text{3-5%}` larger than diameter of rotating geometry and roughly :math:`10%` larger than max height (axis of rotation direction) of rotating geometry
```
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/manualMeshing.rst:73]: fixed

File: docs/source/preprocessing/preprocessing.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (cjdoolittle) [reply docs/source/preprocessing/preprocessing.rst:11]: ah, I see. How about "exported mesh **file** formats"?

[the remark this answers is not in the document]
Comment (jbungenerflex) [reply docs/source/preprocessing/preprocessing.rst]: sounds good.

**Review by maciej-flexcompute — APPROVED** [review]

**Pull request merged**